src/assisted_test_infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -599,6 +600,40 @@ def set_ipxe_url(self, network_name: str, ipxe_url: str):
         self.destroy_network(network)
         self.create_network(xml.toprettyxml())
 
+    def _create_dns_host_xml(self, api_vip: str, cluster_name: str):
+        """
+        Creates an XML objet for adding a DNS host to a libvirt network
+        """
+        host_element = Element("host", ip=api_vip)
+        if cluster_name != "":
+            hostname = f"api.{cluster_name}.{self._entity_config.base_dns_domain}"
+            hostname_element = SubElement(host_element, "hostname")
+            hostname_element.text = hostname
+        return minidom.parseString(tostring(host_element))
+
+    def add_dns_host_to_network(self, network_name: str, api_vip: str, cluster_name: str):
```

Comment:
Can we include the usage for those new functions in this PR?
I think it can help understand a little bit the context, and help us see the testing of it in action

Reply:
I do have it in a separate PR https://github.com/openshift/assisted-test-infra/pull/1816 but should it be included in this one?

Reply:
yes please :pray: 
let's have one PR with the required changes end-to-end

Reply:
Modified this PR to include the changes in the other one! Does this clear it up?

Reply:
hopefully. I'll take a look